src/uct/ib/mlx5/ib_mlx5.inl

```diff
@@ -180,20 +180,26 @@ uct_ib_mlx5_inline_copy(void *restrict dest, const void *restrict src, unsigned
  *
  * @param dest     Inline data in the WQE to copy to.
  * @param iov      A pointer to an array of uct_iov_t elements.
- * @param iov_cnt  A number of elements in a iov array.
+ * @param iov_cnt  A number of elements in iov array.
+ * @param length   A total size of data in iov array.
  * @param wq       Send work-queue.
  */
 static UCS_F_ALWAYS_INLINE void
 uct_ib_mlx5_inline_iov_copy(void *restrict dest, const uct_iov_t *iov,
-                            size_t iovcnt, uct_ib_mlx5_txwq_t *wq)
+                            size_t iovcnt, size_t length,
+                            uct_ib_mlx5_txwq_t *wq)
 {
-    size_t copied;
+    ptrdiff_t remainder;
 
     ucs_assert(dest != NULL);
 
-    copied = uct_iov_to_buffer(iov, iovcnt, 0, dest,
-                               UCS_PTR_BYTE_DIFF(dest, wq->qend));
-    uct_iov_to_buffer(iov, iovcnt, copied, wq->qstart, SIZE_MAX);
+    remainder = UCS_PTR_BYTE_DIFF(dest, wq->qend);
+    if (ucs_likely(length <= remainder)) {
+        uct_iov_to_buffer(iov, iovcnt, 0, dest, SIZE_MAX);
+    } else {
+        uct_iov_to_buffer(iov, iovcnt, 0, dest, remainder);
+        uct_iov_to_buffer(iov, iovcnt, remainder, wq->qstart, SIZE_MAX);
```

Comment:
does it mean we can remove the return value from uct_iov_to_buffer?
or still, better use the return value from first call to uct_iov_to_buffer() as input for the 2nd call, so the compiler would not have to save the previous value of "remainder" and spill more registers to the stack

Reply:
I' think it is better to keep return value.
It will be useful for other tl implementation also.

Reply:
ok